pkg/koordlet/metriccache/metric_cache.go

```diff
@@ -474,6 +570,45 @@ func (m *metricCache) InsertContainerThrottledMetrics(t time.Time, metric *Conta
 	return m.db.InsertContainerThrottledMetric(dbItem)
 }
 
+func (m *metricCache) aggregateGPUUsages(gpuResourceMetricsByTime [][]gpuResourceMetric, aggregateFunc AggregationFunc) ([]*GPUMetric, error) {
+	if len(gpuResourceMetricsByTime) == 0 {
+		return nil, nil
+	}
+	deviceCount := len(gpuResourceMetricsByTime[0])
+	gpuUsageByDevice := make([][]gpuResourceMetric, deviceCount)
```

Comment:
consider using map[gpuid][]gpusMetric as the format of gpuUsageByDevice, in case the gpuid order of deviceMetrics are not always same

Reply:
any suggestions about this?

Reply:
gpusMetric must be sorted by  device minor.